### ISSUE TITLE
accuracy is always zero while loss is decreasing

### ISSUE THREAD

**Issue by Golnoush123** [tensorflow/tensor2tensor#1264]
I have the code below but when I run it I see that all accuracies are zero. Do you know what is the reason?

Epoch 1/5
1029/1029 [==============================] - 0s 191us/step - loss: 1.2047 - acc: 0.0000e+00
Epoch 2/5
1029/1029 [==============================] - 0s 42us/step - loss: 0.1077 - acc: 0.0000e+00
Epoch 3/5
1029/1029 [==============================] - 0s 43us/step - loss: 0.0298 - acc: 0.0000e+00
Epoch 4/5
1029/1029 [==============================] - 0s 40us/step - loss: 0.0192 - acc: 0.0000e+00
Epoch 5/5
1029/1029 [==============================] - 0s 42us/step - loss: 0.0158 - acc: 0.0000e+00

<tensorflow.python.keras.callbacks.History at 0x7fe94328c7b8>

my code:

input = tf.concat([one_hot_vecs0,one_hot_vecs1, one_hot_vecs2, one_hot_vecs3, indices4],1)
print(input.shape) #(1029, 40)
target=np.genfromtxt("basf_targets_rand_normalized.csv", delimiter=',') #t
print(target)
target.shape  #(1029,)
model = keras.Sequential([
    #keras.layers.Flatten(input_shape=(1029, 40)),
    keras.layers.Flatten(),
    keras.layers.Dense(128, activation=tf.nn.relu),
    keras.layers.Dense(1029, activation=tf.nn.softmax)
])
model.compile(optimizer=tf.train.AdamOptimizer(), 
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])
model.fit(data, target, epochs=5)
